src/Connection/StreamConnection.php

```diff
@@ -303,6 +306,13 @@ public function read()
 
         switch ($parsedData['type']) {
             case 'push':
+                $data = [];
+
+                for ($i = 0; $i < $parsedData['value']; ++$i) {
```

Comment:
forach with arr.push?

Reply:
This one defines a behaviour of reader in case if it's Push notification. Since, we're read response line by line, we need to run foreach depends of array.length that we get from server.